phonon/update.py

```diff
@@ -137,37 +138,30 @@ def __setstate__(self, state):
         for k, v in state.items():
             setattr(self, k, v)
 
-    def __cache(self):
+    def __execute(self, block=None):
         """
         Handles deciding whether or not to get the resource from redis. Also
         implements merging cached records with this one (by implementing your
         `merge` method). Increments the number of times this record was
         modified if the cache method executes successfully (does not raise).
         """
-        if self.ref.get_times_modified() > 0:
-            pickled = self.__process.client.get(self.resource_id)
-            if pickled:
-                cached = pickle.loads(pickled)
-                self.merge(cached)
-        self.cache()
-        self.ref.increment_times_modified()
+        raise NotImplemented("Use the Update or ConflictFreeUpdate class instead.")
 
-        if self.init_cache:
-            self.__clear()
-
-    def __execute(self):
+    def __cache(self, block=None):
         """
         Handles deciding whether or not to get the resource from redis. Also
         implements merging cached records with this one (by implementing your
-        `merge` method). Calls your `execute` method to write the record to the
-        database. Recovering from a failed `execute` is up to you.
-        """
-        if self.ref.get_times_modified() > 0:
-            pickled = self.__process.client.get(self.resource_id)
-            if pickled:
-                cached = pickle.loads(pickled)
-                self.merge(cached)
-        self.execute()
+        `merge` method). Increments the number of times this record was
+        modified if the cache method executes successfully (does not raise).
+        """
+        raise NotImplemented("Use the Update or ConflictFreeUpdate class instead.")
+
+    def __merge(self, block=None):
```

Comment:
Private methods are only defined to handle the semantics associated with executing the public ones... like locking in the `Update` class, for example.